src/Sparrow/Json/AbstractBlittableJsonTextWriter.cs

```diff
@@ -453,11 +453,22 @@ private void UnlikelyWriteLargeRawString(byte* buffer, int size)
             var posInStr = 0;
             while (posInStr < size)
             {
-                var amountToCopy = Math.Min(size - posInStr, JsonOperationContext.MemoryBuffer.DefaultSize);
-                FlushInternal();
-                Memory.Copy(_buffer, buffer + posInStr, amountToCopy);
+                var amountToCopy = Math.Min(size - posInStr, JsonOperationContext.MemoryBuffer.DefaultSize - _pos);
+                if (amountToCopy == 0)
+                    goto End; // There is no space available to copy anything, let's just skip and move to flush. 
+
+                Memory.Copy(_buffer + _pos, buffer + posInStr, amountToCopy);
                 posInStr += amountToCopy;
-                _pos = amountToCopy;
+                _pos += amountToCopy;
+
+                // We are not gonna waste a buffer flush if we still have space for other things.
+                // Therefore, we will check if we are done (which is fast) and just break out if
+                // that's the case.
+                if (posInStr == size)
+                    break;
+
+                End:
+                FlushInternal();
```

Comment:
@redknightlois but this is not the same fix that you did here:
https://github.com/ravendb/ravendb/pull/17411/files#diff-d624c45faeb6546cda715cafd343be9520aa9583cf9d77aaa402116d673b79b4L489

The problem you noticed was that we were passing `(uint)buffer.Length);` instead of `(uint)amountToCopy);`

Here (in v5.4) we don't have this issue because this piece of code is different in v5.4 and v6.0 branches:

Current v5.4 impl:

https://github.com/ravendb/ravendb/blob/675159138fabd38dcdad16ef26764e1f59c0079d/src/Sparrow/Json/AbstractBlittableJsonTextWriter.cs#L450-L462

Current v6.0 impl:

https://github.com/ravendb/ravendb/blob/774fbdf7f2f080bd44af839009363ac1902a63f9/src/Sparrow/Json/AbstractBlittableJsonTextWriter.cs#L477-L508

Note that in v6.0 there ate **two** methods `UnlikelyWriteLargeRawString()` - and only one was affected. It was added to v6.0 only in this PR:

https://github.com/ravendb/ravendb/pull/17398/files#diff-d624c45faeb6546cda715cafd343be9520aa9583cf9d77aaa402116d673b79b4R477

Reply:
So there is no issue in v5.4 code and there no need to backport anything because the issue was only in v6.0 code